src/kemal/handler.cr

```diff
@@ -15,6 +15,7 @@ class Kemal::Handler < HTTP::Handler
 
   def add_route(method, path, &handler : Kemal::Context -> _)
     @routes << Route.new(method, path, &handler)
+    @routes << Route.new("HEAD", path, &handler) if method == "GET"
```

Comment:
Singleton `RouteHandler` is not related to this issue, right? You are discussing this from `kemal`'s core architecture point of view?